src/couch/src/couch_doc.erl

```diff
@@ -136,12 +136,27 @@ from_json_obj_validate(EJson, DbName) ->
     case couch_ejson_size:encoded_size(Doc#doc.body) =< MaxSize of
         true ->
              validate_attachment_sizes(Doc#doc.atts),
+             validate_total_document_size(Doc),
              Doc;
         false ->
             throw({request_entity_too_large, Doc#doc.id})
     end.
 
 
+% sum up the json body size + attachment body size and
+% make sure it is < max_http_request_size
+validate_total_document_size(#doc{id=DocId, body=Body, atts=Atts0}) ->
+    MaxReqSize = config:get_integer("httpd", "max_http_request_size", 4294967296), % 2 GB
+    Boundary = couch_uuids:random(), % mock boundary, is only used for the length
```

Comment:
let's hardcode one then if it's just for length.

Reply:
@rnewson I wasn’t sure if the configuration could change the format of the uuid, that’s why I used this one, so we pick that up in current or future uuid format changes.

Happy to chalk this off to YAGNI, just wanted to mention that this was intentional.

Reply:
```
[16:26:10]  <+jan____>	rnewson: do you have a better name for to_tuple in https://github.com/apache/couchdb/pull/1253/files#r178029434?
[16:27:14]  <+rnewson>	I don't, proceed with to_tuple.
```

Reply:
As long as we use uuids, they'd be 32 bytes in size? Why not use  ```<<"00000000000000000000000000000000">>``` ?

Reply:
good call!

Reply:
fixed